packages/sdk/src/evm/core/classes/contract-appuri.ts

```diff
@@ -45,12 +45,14 @@ export class ContractAppURI<TContract extends BaseContract>
    * ```
    * @twfeature AppURI
    */
-  public async get() {
-    if (detectContractFeature<AppURI>(this.contractWrapper, "AppURI")) {
+  public async get(): Promise<string> {
+    if (detectContractFeature<IAppURI>(this.contractWrapper, "AppURI")) {
       return await this.contractWrapper.readContract.appURI();
     }
 
-    return (await this.metadata.get()).appURI || "";
+    const appUri = (await this.metadata.get()).app_uri;
+
+    return Promise.resolve(`ipfs://${appUri.split("/ipfs/")[1]}` || Promise.resolve(""));
```

Comment:
These don't need to be wrapped in `Promise.resolve()` since the function is async.

Also we're assuming that the `appUri` the user passes is an IPFS hash. But they can pass an HTTPS or any other type of URL (like arweave or sia or some other storage system) and this will break. We can't just assume it's IPFS here.